lib/membrane_rtc_engine/engine.ex

```diff
@@ -172,9 +244,39 @@ defmodule Membrane.RTC.Engine do
           node: node()
         ]
 
-  @type notifications :: [Membrane.Element.Action.notify_t()]
+  @typedoc """
+  Membrane action that will cause RTC Engine to publish some message to all other endpoints.
+  """
+  @type publish_action_t() :: {:notify, {:publish, publish_message_t()}}
+
+  @typedoc """
+  Membrane action that make subscribtion for tracks in given format.
 
-  @type tracks_messages :: [{:new_tracks, [Track.t()]} | {:removed_tracks, [Track.t()]}]
+  Endpoint  will be notified about track readiness in `c:Membrane.Bin.handle_pad_added/3` callback.
+  `tracks` is a list in form of pairs `{track_id, track_format}`, where `track_id` is id of track this endpoint subscribes for
+  and `track_format` is the format of track that this endpoint is willing to receive.
+  If `track_format` is `:raw` Endpoint will receive track in `t:#{inspect(__MODULE__)}.Track.encoding/0` format.
+  """
+  @type subscribe_action_t() :: {:notify, {:subscribe, tracks :: [{Track.id(), Track.format()}]}}
+
+  @typedoc """
+  Membrane action that will inform RTC Engine about track readiness.
+  """
+  @type track_ready_action_t() ::
+          {:notify,
+           {:track_ready, Track.id(), Track.encoding(),
+            depayloading_filter :: Membrane.ParentSpec.child_spec_t()}}
+
+  @typedoc """
+  Membrane action that will send Custom Media Event.
+  """
+  @type generate_custom_media_event_action_t() ::
+          {:notify, {:generate_custom_media_event, data :: binary()}}
+
+  @typedoc """
+  Types of messages that can be published to other Endpoints.
+  """
+  @type publish_message_t() :: {:new_tracks, [Track.t()]} | {:removed_tracks, [Track.t()]}
```

Comment:
I am wondering if we shouldn't covert these two into Message structs :thinking:

Reply:
The only advantage I see in this type is that we have gathered all types of messages that can be published in one place. But with the use of custom endpoints, it became useless. With having it in mind I think we should add a possibility to publish any message to other endpoints.